src/snowflake/connector/ocsp_snowflake.py

```diff
@@ -1410,16 +1420,22 @@ def _fetch_ocsp_response(
         headers = {HTTP_HEADER_USER_AGENT: PYTHON_CONNECTOR_USER_AGENT}
 
         if not OCSPServer.is_enabled_new_ocsp_endpoint():
+            # this is the core logic, get and post will construct different pay loads
             actual_method = "post" if self._use_post_method else "get"
             if self.OCSP_CACHE_SERVER.OCSP_RETRY_URL:
                 # no POST is supported for Retry URL at the moment.
                 actual_method = "get"
 
             if actual_method == "get":
+                # the logic here feels weird, `actual_method` is `post`, but we are
+                # generating a "get" url and set payload to None which feels more like this is a "get" op?
+                # shouldn't the get and post be flipped?
                 b64data = self.decode_ocsp_request_b64(ocsp_request)
                 target_url = self.OCSP_CACHE_SERVER.generate_get_url(ocsp_url, b64data)
                 payload = None
             else:
+                # would enforcing "get" rather than "post" work?
+                # can try it by flipping the flag 'use_post_method' in ssl_wrap_socket
                 target_url = ocsp_url
                 payload = self.decode_ocsp_request(ocsp_request)
                 headers["Content-Type"] = "application/ocsp-request"
```

Comment:
is there a chance that we haven't tested this logic before at all? it's always broken, just so far not hit by any customer?